File: examples/deadline/EC2-Image-Builder/ts/lib/base-farm-stack.ts
/**
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import {
  Vpc,
} from 'aws-cdk-lib/aws-ec2';
import {
  RemovalPolicy,
  Stack,
  StackProps
} from 'aws-cdk-lib';
import {
  AwsCustomerAgreementAndIpLicenseAcceptance,
  RenderQueue,
  Repository,
  ThinkboxDockerImages,
  VersionQuery
} from 'aws-rfdk/deadline';
import { Construct } from 'constructs';

export interface FarmProps extends StackProps {
  /**
   * Whether the AWS Customer Agreement and AWS Intellectual Property License are agreed to.
   */
  readonly userAwsCustomerAgreementAndIpLicenseAcceptance: AwsCustomerAgreementAndIpLicenseAcceptance;

  /**
   * Version of Deadline to use.
   */
  readonly deadlineVersion: string;
}

/**
 * This stack includes all the basic setup required for a render farm. It excludes the worker fleet.
 */
export class BaseFarmStack extends Stack {
  public readonly renderQueue: RenderQueue;
  public readonly vpc: Vpc;

  constructor(scope: Construct, id: string, props: FarmProps) {
    super(scope, id, props);

    this.vpc = new Vpc(this, 'Vpc', { maxAzs: 2 });

    const version = new VersionQuery(this, 'Version', {
      version: props.deadlineVersion,
    });

    const images = new ThinkboxDockerImages(this, 'Images', {
      version: version,
      userAwsCustomerAgreementAndIpLicenseAcceptance: props.userAwsCustomerAgreementAndIpLicenseAcceptance,
    });

    const repository = new Repository(this, 'Repository', {
      vpc: this.vpc,
      removalPolicy: {
        database: RemovalPolicy.DESTROY,
        filesystem: RemovalPolicy.DESTROY,
      },
      version,
    });

    this.renderQueue = new RenderQueue(this, 'RenderQueue', {
      vpc: this.vpc,
      version,
      images,
      repository,
      deletionProtection: false,
    });
  }
}

File: examples/deadline/EC2-Image-Builder/python/package/lib/base_farm_stack.py
# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
# SPDX-License-Identifier: Apache-2.0

from dataclasses import dataclass

from aws_cdk import (
    RemovalPolicy,
    Stack,
    StackProps
)
from aws_cdk.aws_ec2 import (
    Vpc,
)
from aws_rfdk.deadline import (
    AwsCustomerAgreementAndIpLicenseAcceptance,
    RenderQueue,
    Repository,
    RepositoryRemovalPolicies,
    ThinkboxDockerImages,
    VersionQuery,
)
from constructs import (
    Construct
)


@dataclass
class BaseFarmStackProps(StackProps):
    # Whether the AWS Customer Agreement and AWS Intellectual Property License are agreed to.
    accept_aws_customer_agreement_and_ip_license: AwsCustomerAgreementAndIpLicenseAcceptance

    # Version of Deadline to use
    deadline_version: str


class BaseFarmStack(Stack):
    """
    This stack includes all the basic setup required for a render farm. It excludes the worker fleet.
    """

    def __init__(self, scope: Construct, stack_id: str, *, props: BaseFarmStackProps, **kwargs):
        """
        Initialize a new instance of BaseFarmStack
        """
        super().__init__(scope, stack_id, **kwargs)

         # The VPC that all components of the render farm will be created in.
        self.vpc = Vpc(
            self,
            'Vpc',
            max_azs=2,
        )

        version = VersionQuery(
            self,
            'Version',
            version=props.deadline_version,
        )

        images = ThinkboxDockerImages(
            self,
            'Images',
            version=version,
            user_aws_customer_agreement_and_ip_license_acceptance=props.accept_aws_customer_agreement_and_ip_license,
        )

        repository = Repository(
            self,
            'Repository',
            removal_policy=RepositoryRemovalPolicies(
                database=RemovalPolicy.DESTROY,
                filesystem=RemovalPolicy.DESTROY,
            ),
            vpc=self.vpc,
            version=version,
        )

        self.render_queue = RenderQueue(
            self,
            'RenderQueue',
            vpc=self.vpc,
            version=version,
            images=images,
            repository=repository,
            deletion_protection=False,
        )
